All return types in interfaces must be operation results

realpj10101 opened the pull request:
Modified all interfaces, repos and controller with op result

realpj10101 committed:
Changed IAcRepo return types to op result

realpj10101 committed:
Changed AcRepo return types to op result

realpj10101 committed:
Added more error codes

realpj10101 committed:
Added more error codes

realpj10101 committed:
Modified AcController with op result

realpj10101 committed:
Changed IAdminRepo return type to op result

realpj10101 committed:
Changed AdminRepo return type to op result

realpj10101 committed:
Modified admin controller with op result

realpj10101 committed:
Added more error codes

realpj10101 committed:
Added more error codes

realpj10101 committed:
Changed site site interface return types to op result

realpj10101 committed:
Changes sote repo return types to op result

realpj10101 committed:
Modified site controller with op result

realpj10101 committed:
Changed total  minutes to total hours

realpj10101 committed:
Changed course interface return type to op result

realpj10101 committed:
Changed course interface return type to op result

realpj10101 committed:
Changed course repo return type to op result

realpj10101 committed:
Modified course controller by op result

hrahiminasr committed:
redirect to courses page on cancel button in add new course (#521)

vahid-hayati committed:
Renamed 'Class' entity to 'ClassRoom' to fix reserved keyword conflict (#522)

realpj10101 committed:
Changed class interfaces return type to op result

realpj10101 committed:
Changed class repo return types to op result

realpj10101 committed:
Modified class controller with op result

realpj10101 committed:
Changed manager interface return types to op result

realpj10101 committed:
Changed manager repo return types to op result

realpj10101 committed:
Changed manager repo return types to op result

realpj10101 committed:
Modified manager controller with op result

realpj10101 committed:
Modified manager controller with op result

realpj10101 committed:
Modified mapper

realpj10101 committed:
Added mroe error codes

realpj10101 committed:
Added mroe error codes

realpj10101 committed:
Changed member interface return types to op result

realpj10101 committed:
Changed member repo return types to oip result

realpj10101 committed:
Changed member repo return types to oip result

realpj10101 committed:
Modified member controller by op result

realpj10101 committed:
Modified member controller by op result

realpj10101 committed:
Changed teacher interface return types with op result

realpj10101 committed:
Changed teacher repo return types with op result

realpj10101 committed:
Changed teacher repo return types with op result

realpj10101 committed:
Modified teacher controller with op result

realpj10101 committed:
Modified teacher controller with op result

realpj10101 committed:
Changed check is absent interface return type to op result

realpj10101 committed:
Changed is absent return type to op result

FaranakBaran commented on backend/api/DTOs/Helpers/ErrorCode.cs:
```diff
@@ -7,12 +7,26 @@ public enum ErrorCode
     IsCourseNotFound,
     IsUserNotFound,
     IsSiteNotFound,
+    IsClasssNotFound,
```
typo: IsClassNotFound

realpj10101 replied:
Done

FaranakBaran commented on backend/api/DTOs/CourseDto.cs:
```diff
@@ -6,8 +6,8 @@ public record CreateCourseDto(
     string Description,
-    [Required, Range(0.5, 20000, ErrorMessage = "ساعت دوره باید بین ۰٫۵ ساعت تا ۲۰,۰۰۰ ساعت باشد")]
-    int TotalMinutes,
+    [Required, Range(0.5, 200, ErrorMessage = "ساعت دوره باید بین ۰٫۵ ساعت تا ۲۰۰ ساعت باشد")]
```
تا جایی که یادمه حداکثر ساعت دوره باید ۲۰۰۰ ساعت باشه و  ۲۰۰ فک میکنم کمه

realpj10101 replied:
Done

FaranakBaran commented on backend/api/Repositories/TeacherRepository.cs:
```diff
@@ -65,42 +83,49 @@ public async Task<List<ClassRoom>> GetClassesAsync(string hashedUserId, Cancella
-        if (ObjectId.Equals != null)
-        {
-            ShowStudentStatusDto showStudentStatusDto = Mappers.ConvertAttendenceToShowStudentStatusDto(attendence);
+        Attendance? attendence = Mappers.ConvertAddStudentStatusDtoToAttendence(teacherInput, targetAppUser.Id, targetCourseId, currentDate);
```
typo: attendance

realpj10101 replied:
Done

FaranakBaran commented on backend/api/Repositories/TeacherRepository.cs:
```diff
@@ -65,42 +83,49 @@ public async Task<List<ClassRoom>> GetClassesAsync(string hashedUserId, Cancella
+                false,
+                Error: new(
+                    ErrorCode.IsAlreadyEnrolled,
+                    "User is already a attendace"
```
attendance

realpj10101 replied:
Done

realpj10101 replied:
Done

FaranakBaran commented on backend/api/Repositories/TeacherRepository.cs:
```diff
@@ -65,42 +83,49 @@ public async Task<List<ClassRoom>> GetClassesAsync(string hashedUserId, Cancella
 
-            return showStudentStatusDto;
-        }
+        await _collectionAttendence.InsertOneAsync(attendence, null, cancellationToken);
```
attendance

realpj10101 replied:
Done

FaranakBaran commented on backend/api/Repositories/TeacherRepository.cs:
```diff
@@ -65,42 +83,49 @@ public async Task<List<ClassRoom>> GetClassesAsync(string hashedUserId, Cancella
-        return null;
+        return new(
+            true,
+            Mappers.ConvertAttendenceToShowStudentStatusDto(attendence),
```
attendance

realpj10101 replied:
Done

FaranakBaran commented on backend/api/Controllers/ManagerController.cs:
```diff
@@ -58,15 +61,18 @@ public async Task<ActionResult<RegisteredUserDto>> CreateStudent(RegisterDto man
+            ErrorCode.IsDuplicatePhone => BadRequest(opResult.Error.Message),
+            ErrorCode.IsIdentityFailed => BadRequest(opResult.Error.Message),
+            ErrorCode.IsRoleIdentityFailed => BadRequest(opResult.Error.Message),
+            _ => BadRequest("Opertion failed! Try again or contact support.")
```
Operation

realpj10101 replied:
Done

FaranakBaran replied:
Operation

FaranakBaran commented on backend/api/Controllers/ManagerController.cs:
```diff
@@ -373,23 +436,33 @@ public async Task<ActionResult<IEnumerable<ShowStudentStatusDto>>> GetAllAttende
         List<ShowStudentStatusDto> showStudentStatusDtos = [];
 
-        foreach (Attendance attendence in pagedAttendences)
+        foreach (Attendance attendence in opResult.Result)
```
attendance

realpj10101 replied:
Done

realpj10101 committed:
FIxed error code type error

realpj10101 committed:
FIxed error code type error

FaranakBaran commented on backend/api/Controllers/MemberController.cs:
```diff
@@ -32,23 +36,33 @@ public async Task<ActionResult<IEnumerable<ShowStudentStatusDto>>> GetAllAttende
         List<ShowStudentStatusDto> showStudentStatusDtos = [];
 
-        foreach (Attendance attendence in pagedAttendences)
+        foreach (Attendance attendence in opResult.Result)
```
attendance

realpj10101 replied:
Done

FaranakBaran commented on backend/api/Repositories/MemberRepository.cs:
```diff
@@ -17,40 +17,54 @@ public MemberRepository(IMongoClient client, IMyMongoDbSettings dbSettings, ITok
 
-        return await PagedList<Attendance>.CreatePagedListAsync(query, attendenceParams.PageNumber, attendenceParams.PageSize, cancellationToken);
+        PagedList<Attendance> attendances = await PagedList<Attendance>.CreatePagedListAsync(query, attendenceParams.PageNumber, attendenceParams.PageSize, cancellationToken);
+
```
attendanceParams

realpj10101 replied:
Done

realpj10101 committed:
Changed range ftom 200 to 2000

realpj10101 committed:
Fixed typo

realpj10101 committed:
Fixed typo

realpj10101 committed:
Fixed typo

realpj10101 committed:
Fixed typo

realpj10101 committed:
Fixed typo

realpj10101 committed:
Fixed typo

vahid-hayati committed:
Validated DeleteResult and UpdateResult in all repository methods (#523)

realpj10101 committed:
feat(api): add delete classroom interface

Classroom module does not have did not have delete feature. Create the interface

realpj10101 committed:
feat(api): add delete repo

Classroom module does not have did not have delete feature. Create the repo

realpj10101 committed:
feat(api): add delete controller

Classroom module does not have did not have delete feature. Create the controller

realpj10101 committed:
feat(api): add error codes

For validate delete class room errors, add more erro codes.

realpj10101 committed:
Merge branch 'develop' into feature/backend/511-all-return-types-in-interfaces-must-be-operation-results

realpj10101 committed:
refactor(api):  use operation result for classroom interface return types

realpj10101 committed:
refactor(api): unify repository return types using operation result

Update repository implementation to handle success and error states

realpj10101 committed:
refactor(api): oprimize classroom controller using operation result

realpj10101 committed:
refactor(api): unify manager interface return types using operation result

Refactor manager interface return types for consistency

realpj10101 committed:
refactor(api): unify manager repo return types using operation result

realpj10101 committed:
refactor(api): optimize manager controller using operation result

Response of controller was not accurate use operation ressult for optimization

realpj10101 committed:
refactor(api): unify  member interface using operation result

Refactor member interfaces for consistency

realpj10101 committed:
refactor(api): unify member repo return types using operation result

Refactor member repo return types for consistency

realpj10101 committed:
refactor(api): optimize member controller using operation result

Member controller responses was not accurate make it optimize

realpj10101 committed:
refactor(api): unify teacher interface return types using  operation result

Refactor reacher interface return types for consistency

realpj10101 committed:
refactor(api): unify teacher repo return types using operation result

Refactor teacher repo for consistency

realpj10101 committed:
refactor(api): optimize teacher controller with operation result

Teacher controller responses were not accurate make it optimize

realpj10101 committed:
refactor(api): use IEnumerable instead of list

Use IEnumerable instead of  list for better performance

realpj10101 committed:
refactor(api): remove is class not found from error codes

realpj10101 committed:
refactor(api): use classroom error code

For greater readability use class room error code

realpj10101 committed:
refactor(api): use classroom error code

For greater readablity user classroom

realpj10101 committed:
refactor(api): user classroom error code

For readability user classroom error codes

realpj10101 committed:
refactor(api): use classroom error code

For greater readabilitu user classroom error code

FaranakBaran commented on backend/api/Repositories/ManagerRepository.cs:
```diff
@@ -818,7 +875,7 @@ CancellationToken cancellationToken
         false,
         Error: new(
-          ErrorCode.IsClassNotFound,
+          ErrorCode.IsClasssNotFound,
```
IsClassNotFound

FaranakBaran commented on backend/api/Controllers/ManagerController.cs:
```diff
@@ -71,7 +71,7 @@ public async Task<ActionResult<RegisteredUserDto>> CreateStudent(RegisterDto man
             ErrorCode.IsIdentityFailed => BadRequest(opResult.Error.Message),
             ErrorCode.IsRoleIdentityFailed => BadRequest(opResult.Error.Message),
-            _ => BadRequest("Operation failed! Try again or contact support.")
+            _ => BadRequest("Opertion failed! Try again or contact support.")
```
Operation

realpj10101 replied:
Done

FaranakBaran commented on backend/api/Repositories/MemberRepository.cs:
```diff
@@ -24,7 +24,7 @@ public MemberRepository(IMongoClient client, IMyMongoDbSettings dbSettings, ITok
     #endregion Constructor
 
-    public async Task<OperationResult<PagedList<Attendance>>> GetAllAttendenceAsync(AttendenceParams attendanceParams, ObjectId? userId, string targetClassTitle, CancellationToken cancellationToken)
+    public async Task<OperationResult<PagedList<Attendance>>> GetAllAttendenceAsync(AttendenceParams attendenceParams, ObjectId? userId, string targetClassTitle, CancellationToken cancellationToken)
```
AttendanceParams

realpj10101 replied:
Done

FaranakBaran commented on backend/api/Repositories/MemberRepository.cs:
```diff
@@ -58,7 +58,7 @@ public async Task<OperationResult<PagedList<Attendance>>> GetAllAttendenceAsync(
             .Where(doc => doc.StudentId == appUser.Id && doc.ClassId == targetClassId);
 
-        PagedList<Attendance> attendances = await PagedList<Attendance>.CreatePagedListAsync(query, attendanceParams.PageNumber, attendanceParams.PageSize, cancellationToken);
+        PagedList<Attendance> attendances = await PagedList<Attendance>.CreatePagedListAsync(query, attendenceParams.PageNumber, attendenceParams.PageSize, cancellationToken);
```
AttendanceParams

realpj10101 replied:
Done

FaranakBaran commented on backend/api/Repositories/TeacherRepository.cs:
```diff
@@ -65,63 +83,88 @@ public async Task<List<ClassRoom>> GetClassesAsync(string hashedUserId, Cancella
-        if (ObjectId.Equals != null)
-        {
-            ShowStudentStatusDto showStudentStatusDto = Mappers.ConvertAttendenceToShowStudentStatusDto(attendence);
+        Attendance? attendance = Mappers.ConvertAddStudentStatusDtoToAttendence(teacherInput, targetAppUser.Id, targetCourseId, currentDate);
```
ConvertAddStudentStatusDtoToAttendance

realpj10101 replied:
Done

FaranakBaran commented on backend/api/Controllers/ManagerController.cs:
```diff
@@ -373,25 +436,35 @@ public async Task<ActionResult<IEnumerable<ShowStudentStatusDto>>> GetAllAttende
+        foreach (Attendance attendence in opResult.Result)
         {
-            showStudentStatusDtos.Add(Mappers.ConvertAttendenceToShowStudentStatusDto(attendence));
+            showStudentStatusDtos.Add(Mappers.ConvertAttendanceToShowStudentStatusDto(attendence));
```
attendance

realpj10101 replied:
Done

realpj10101 committed:
refactor(api): fix typo

Attendance has typo error

realpj10101 committed:
refactor(api): fix typo

Attendance has type error

realpj10101 committed:
refactor(api): fix error typo

Operation has typo error

realpj10101 committed:
refactor(api): fix typo

Attendace has typo error

FaranakBaran commented:
![Uploading Screenshot_2025-12-28_03-43-38.png…]()

FaranakBaran commented:
![Uploading Screenshot_2025-12-28_03-43-38.png…]()

FaranakBaran committed:
Merge branch 'develop' into feature/backend/511-all-return-types-in-interfaces-must-be-operation-results

FaranakBaran commented:
<img width="585" height="1266" alt="IMG_7705" src="https://github.com/user-attachments/assets/2cc1d65b-8be6-4021-ac12-75daf1fd3ae8" />

FaranakBaran commented:
![IMG_7704](https://github.com/user-attachments/assets/599833f9-6a60-4ba1-8523-44a8ce847257)

realpj10101 committed:
refactor(api): remove duplicate method

In rebases delete classroom written twice

realpj10101 committed:
fix(api): fix warnings

There was some warning on return types

FaranakBaran reviewed (approved):
Done
